Library/Homebrew/cmd/search.rb

```diff
@@ -41,6 +41,8 @@ def search_args
              description: "Search online and locally for formulae."
       switch "--cask", "--casks",
              description: "Search online and locally for casks."
+      flag "--deleted=",
+           description: "Search for a formula deleted after a given date (default: `1 month ago`)."
```

Comment:
How about adding this flag but defaulting to "forever"?

Reply:
When you say "default", do you mean "default without `--deleted`", or "default when `--deleted` is used without a date"?

Reply:
I think defaulting to forever when `--deleted` is used without a date would be good. Amending the existing message about searching for deleted formulae to suggest using `--deleted` to search a longer horizon might be helpful.

Reply:
> I think defaulting to forever when `--deleted` is used without a date would be good.

Yup, this 👍🏻